docs/src/reference-guide/components/graphql/directives.md

```diff
@@ -65,3 +66,22 @@ type Library {
 ```
 
 A foreign key constraint will be created on `library.book` that references `book.name`, which relates the `Book`s in a `Library` to the underlying `Book` table.
+
+## `@virtual`
+
+The `@virtual` directive instructs the indexer's SQL schema builder to _not_ build SQL tables from types that include this directive on any field.
+
+```graphql
+type Title {
+    name: CharField! @virtual
+}
+
+type Book {
+    id: ID!
+    title: Title!
+}
+```
+
+When SQL tables are generated for the entities above, a table will be created for `Book`, but no table will be created for `Title`. Rather, the `title` field on the `Book` object will exist on the `book` table as a `JSON` field.
+
+> Important: When using the `@virtual` directive with GraphQL `union` types, each member of the `union` type must either include _only_ types that are not virtual, or _only_ types that are virtual. We do not support mixing and matching virtual types with non-virtual types in unions.
```

Comment:
I think it would be helpful to include an example of when a user may want to mark a field with `@virtual`.

Reply:
@deekerno Will add this to the to-do list